src/Microsoft.AspNet.TestHost/ClientHandler.cs

```diff
@@ -129,7 +121,7 @@ internal RequestState(HttpRequestMessage request, PathString pathBase, IHttpCont
                     request.Headers.Host = request.RequestUri.GetComponents(UriComponents.HostAndPort, UriFormat.UriEscaped);
                 }
 
-                HttpContext = _factory.Create(new FeatureCollection());
+                HttpContext = new DefaultHttpContext();
```

Comment:
This should use IHttpApplication. Here's another case where we would probably want to keep application as an `object` since it will need to be passed to `ClientHandler` and `WebSocketClient` to handle create/processes/dispose.